kubernetes/cache/cache.go

```diff
@@ -149,6 +151,34 @@ func NewKialiCache() (KialiCache, error) {
 	kialiCacheImpl.k8sApi = istioClient.GetK8sApi()
 	kialiCacheImpl.istioApi = istioClient.Istio()
 
+	// Update SA Token
+	ticker := time.NewTicker(tokenExpireDuration)
+	quit := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				if newToken, err := kubernetes.GetKialiToken(); err != nil {
+					log.Errorf("Error updating Kiali Token %v", err)
+				} else {
+					log.Debug("Kiali Cache: Updating token")
+					istioConfig.BearerToken = newToken
+					var errorInitClient error
+					istioClient, errorInitClient = kubernetes.NewClientFromConfig(&istioConfig)
+					if errorInitClient != nil {
+						log.Errorf("Error creating new Client From Config %v", errorInitClient)
+					}
+					kialiCacheImpl.istioClient = *istioClient
+					kialiCacheImpl.k8sApi = istioClient.GetK8sApi()
+					kialiCacheImpl.istioApi = istioClient.Istio()
+				}
+			case <-quit:
+				ticker.Stop()
+				return
```

Comment:
I'm wondering if this is dead code, as I don't understand where the `quit` channel is fed.

Reply:
Yes you want to close the `quit` channel in the `Stop` method. That's the only way this goroutine will exit.

Reply:
Added the change in the Stop method.